### PR TITLE
Move meta classifiers

### PR DESCRIPTION
**Pull request by t-tte**
Added classification capability to the base-learners as the first step.

### CODE REVIEW

**Comment by jeongyoonlee** [comment]
Thanks for taking the lead on this, @t-tte. I think it might be better to extend base meta-learner classes to base meta-classifier and meta-regressor classes similar to how scikit-learn does?

**Comment by t-tte** [comment]
@jeongyoonlee - Just pushed the version with meta-classifiers and meta-regressors as separate classes.

File: causalml/inference/meta/slearner.py [hunk 1]
```diff
@@ -223,3 +198,131 @@ def estimate_ate(self, X, treatment, y):
...
+                mae(y[~is_treatment], yhat_c[~is_treatment]))
+            )
+            logger.info('RMSE (Treatment): {:.6f}'.format(
+                np.sqrt(mse(y[is_treatment], yhat_t[is_treatment])))
+            )
+            logger.info(' MAE (Treatment): {:.6f}'.format(
+                mae(y[is_treatment], yhat_t[is_treatment]))
+            )
+
+        return (yhat_t - yhat_c).reshape(-1, 1)
```
Comment (yungmsh) [inline causalml/inference/meta/slearner.py]: We might want to add np.clip here to ensure values are between 0 and 1 (e.g. if yhat_c > yhat_t)

**Comment by t-tte** [comment]
As discussed, leaved the BaseLearner classes as they are and extended to Regressor and Classifier subclasses per each meta-learner.

**Comment by jeongyoonlee** [comment]
Thanks! Code looks great. Could you please update the tests for meta-learners with new API change? Also what about leaving `Base[S,X,T,R]Learner` in `inference.meta.__init__.py` for backward compatibility?

**Comment by t-tte** [comment]
Both sound good, will push changes later today.

**Comment by t-tte** [comment]
Tests have been updated.

**Comment by jeongyoonlee** [comment]
Thanks @t-tte. Test code looks good as well. One fix needed: Could you please change `LRSLearner` to `LRSRegressor` in `tests/test_metrics.py` to the build pass the test (https://travis-ci.com/uber/causalml/jobs/227749855#L562)

**Pull request merged**